src/jit/gentree.cpp

```diff
@@ -611,6 +611,60 @@ void Compiler::fgWalkAllTreesPre(fgWalkPreFn* visitor, void* pCallBackData)
 }
 
 // ------------------------------------------------------------------------------------------
+// gtCheckReg: Checks that the register has either been set, or is still set to the "no register
+// assignment" value, depending upon whether it fits any of the special cases for multi-reg call nodes.
+//
+// Arguments:
+//     compiler  -  compiler instance
+//
+// Return Value:
+//     None
+//
+// Notes:
+//     This also sets the gtRegTag to GT_REGTAG_REG.
+//
+void GenTree::gtCheckReg()
+{
+#ifdef DEBUG
+    assert((gtRegTag != GT_REGTAG_NONE) || (gtRegNum == REG_NA));
+#if CPU_LONG_USES_REGPAIR
+    assert(!"gtSetRegDefault not supported for REGPAIR");
+#endif // CPU_LONG_USES_REGPAIR
+    if (gtRegTag == GT_REGTAG_NONE)
+    {
+        gtRegTag = GT_REGTAG_REG;
+
```

Comment:
Maybe, if the `gtRegTag` facility is not providing value for RyuJIT / LSRA (it mostly provided value on LEGACY_BACKEND to prevent confusion between gtRegNum and gtRegPair), we should simply stop using it. One way (possibly) would be to change the GenTree constructor from:
```
    gtRegNum = REG_NA;
    INDEBUG(gtRegTag = GT_REGTAG_NONE;)
```
to:
```
    gtRegNum = REG_NA;
#ifdef LEGACY_BACKEND
    INDEBUG(gtRegTag = GT_REGTAG_NONE;)
#else
    INDEBUG(gtRegTag = GT_REGTAG_REG;) // RyuJIT does not use the gtRegTag assertions
#endif
```
Another way would be to put the relevant asserts under `#ifdef LEGACY_BACKEND`.